packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -170,7 +174,52 @@ export default class Editor {
      * @returns true if node is inserted. Otherwise false
      */
     public insertNode(node: Node, option?: InsertOption): boolean {

```

Comment:
curious why we opted to implement this at the Editor vs the EditorCore level.

The distinction isn't clear to me, besides the fact that the consumer can override the core API.

Reply:
Yeah, you're right, this could live at either level. SetContent needs to remain at the editor level, however (as there's no corresponding core api for it) Defer to Jason for more context about this.

Reply:
I don't want to block this check in, so for now I feel it is ok to put the code here.

But as a long term thinking, I'd like to keep everything about dark mode in a small scope, say, a plugin.

My basic idea is, for everything which needs taking care of color (such as set/getContent, setText/BkColor, Paste, ...), we can fire an event before/after the content is changed, and let the plugin handle everything about color change. So we don't need to put any dark specified code everywhere. And we already have ContentChangedEvent today, but if that event is not sufficient, we can add a new event.

And this can happen when we finally integrate the code back to master.

Reply:
I know that's your position, but we run into problems with colors and guaranteeing execution order. I'm not pleased with the "this plugin HAS to be last" because it doesn't feel like that's enforceable, but that's the only way to make any guarantees about the content. Let's come up with a better strategy for this (core API override?) before going straight to plugin.

Reply:
core API override is also an option.

About your concern of plugin order, there are two options:
1. Make it as a core plugin, so we can decide the order in editor code, and user won't see it.
2. Since we are adding a new event just for color handling, other plugins are not likely to handle it, so it doesn't matter in what order we add the plugin.

Reply:
Core plugin, we still have to guarantee that the plugin list never changes, and then we end up special casing the plugin order, which still leaves a lot of dark mode specific code, just in the plugin init.

As for 2, just because it's not likely to be used doesn't mean that it _won't_ be, and obscurity isn't a good enough guarantee for this.